File: Palindrome Number/Palindrome Number.java
public class Solution {
    public boolean isPalindrome(int x) {
        if(x < 0)               return false;
        else if(x == 0)         return true;
        
        long lMask = 1;
        long rMask = 1;
        long temp = x;
        
        while(temp > 0){
            temp /= 10;
            lMask *= 10;
        }
        lMask /= 10;
        
        while(lMask > rMask){
            long msb = (x / lMask) % 10;
            long lsb = (x % (rMask * 10)) / rMask;
            if(msb != lsb){
                return false;
            }
            lMask /= 10;
            rMask *= 10;
        }
        
        return true;
    }
}

File: Palindrome Number/Palindrome Number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x < 0)   return false;
        int tmp = x;
        int counter = 0;
        while(tmp){
            ++counter;
            tmp /= 10;
        }
        
        while(x){
            int mask = 0;
            int count = 0;
            while(count++ < counter){
                if(mask)    mask *= 10;
                else        mask = 1;
            }
            
            int MSB = x / mask;
            int LSB = x % 10;
            if(MSB != LSB)      return false;
            x -= MSB * mask;
            if(!x)      break;
            x /= 10;
            counter -= 2;
        }
        return true;
    }
};
